### PR TITLE
Login fixes

### PR DESCRIPTION
**Pull request by Doodshoofdaapje**

### CODE REVIEW

**Review by marnikdenouden — APPROVED** [review]
I think it is good

File: backend/api/views/leaderboard_entry_view.py [hunk 1]
```diff
@@ -4,25 +4,26 @@
...
-    def get(self, request, submission_id:str) -> JsonResponse:
+    def get(self, request, problem_id, submission_id:str) -> JsonResponse:
         """Get request to retrieve a leaderboard entry for a submission id."""
 
         # Get the submission from the id
         submission = Submission.objects.all().filter(id=submission_id)
+        problem = SpecifiedProblem.objects.all().filter(id=problem_id)
 
         # Robustness check to see if submission exists
         if not submission.exists():
```
Comment (TPGamesNL) [inline backend/api/views/leaderboard_entry_view.py:21]: Also add a similar check for problem

File: backend/api/views/submit_view.py [hunk 2]
```diff
@@ -176,20 +179,20 @@ def get(self, request, sidb64, token):
         except ObjectDoesNotExist:
             self.logger.warning("Could not locate user")
             return Response(
-                {"User error": "User not found."}, status=status.HTTP_400_BAD_REQUEST
+                {"detail": "User not found."}, status=status.HTTP_400_BAD_REQUEST
             )
+        
+        self.logger.warning(submission)
```
Comment (TPGamesNL) [inline backend/api/views/submit_view.py:186]:
```suggestion

```

File: backend/tests/test_auth_view.py [hunk 3]
```diff
@@ -37,11 +48,13 @@ def test_sign_up_valid(self):
         self.auth.signup(req)
         
         #Extract token and uid from the mail that was sent by the signup method
-        link_index = mail.outbox[0].body.find('http://testserver/api/activate/')
+        link_index = mail.outbox[0].body.find(os.getenv("FRONTEND_URL") + "/verify")#'http://testserver/api/activate/')
```
Comment (TPGamesNL) [inline backend/tests/test_auth_view.py:51]:
```suggestion
        link_index = mail.outbox[0].body.find(os.getenv("FRONTEND_URL") + "/verify")
```

File: frontend/src/main.scss
Comment (TPGamesNL) [inline frontend/src/main.scss:1]: Undo

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**